### PR TITLE
Redirect user to home on APRC failures

### PR DESCRIPTION
**Pull request by yotommy**
### Description

In cases where the `ApplicantProgramReviewController` cannot fetch the user profile, or the user is unauthorized, redirect the user to the home page.

## Release notes

In cases where the user profile cannot be fetched, or the user is unauthorized, redirect the user to the home page. This case could occur if a tab is left open overnight and the browser session expires.

If a user profile is not available, then it doesn't make sense to create a guest browser session and try the action again, since the new account will not be authorized to see the application for the existing account. Sending the user to the homepage seems reasonable for this case.

Similarly, if the user fails authorization for any reason, we currently just return 401 and nothing shows in the browser. Redirecting to the home page also makes sense here.

This addresses two out of three stack traces reported in #4944.

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: server/test/controllers/HasBooleanHeaderArgumentMatcher.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class HasBooleanHeaderArgumentMatcher implements ArgumentMatcher<Http.Request> {
+  private final String headerName;
+
+  HasBooleanHeaderArgumentMatcher(String headerName) {
+    this.headerName = headerName;
+  }
+
+  @Override
+  public boolean matches(Http.Request argument) {
+    if (argument == null || !argument.hasHeader(headerName)) return false;
```
Comment (bion) [inline server/test/controllers/HasBooleanHeaderArgumentMatcher.java:18]: **nitpick:** https://google.github.io/styleguide/javaguide.html#s4.1-braces
```suggestion
    if (argument == null || !argument.hasHeader(headerName)) {
      return false;
    }
```

File: server/test/controllers/HasBooleanHeaderArgumentMatcher.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package controllers;
+
+import org.mockito.ArgumentMatcher;
+import play.mvc.Http;
+
+/**
+ * ArgumentMatcher that matches an Http.Request containing a specified header whose value is "true".
+ */
+public class HasBooleanHeaderArgumentMatcher implements ArgumentMatcher<Http.Request> {
```
Comment (bion) [inline server/test/controllers/HasBooleanHeaderArgumentMatcher.java:9]: **nitpick (blocking):** 
```suggestion
public final class HasBooleanHeaderArgumentMatcher implements ArgumentMatcher<Http.Request> {
```

**Review by bion — APPROVED** [review]
two small items, otherwise looks good

**Comment by yotommy** [comment]
The updated [coverage report](https://app.codecov.io/github/civiform/civiform/commit/3fc566f35c13c38136dbfc6b25b69feb3a85606e/blob/server/app/controllers/applicant/ApplicantProgramReviewController.java) shows the new code paths are covered by tests.

**Pull request merged**